Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java

```diff
@@ -0,0 +1,84 @@
+package nl.soccar.gameserver.model;
+
+import nl.soccar.gameserver.controller.socnet.message.ChatMessage;
+import nl.soccar.gameserver.model.command.Command;
+import nl.soccar.gameserver.model.command.CommandDispatcher;
+import nl.soccar.gameserver.model.session.RoomWrapper;
+import nl.soccar.library.enumeration.Privilege;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * @author Lesley
+ */
+public final class Chat {
+
+    private final List<String> mutedNames = new ArrayList<>();
+
+    private final RoomWrapper room;
+
+    public Chat(RoomWrapper room) {
+        this.room = room;
+    }
+
+    public void sendChatMessage(PlayerWrapper player, String message) {
+        if (message.startsWith("/")) {
+            String[] components = message.substring(1).split(" ");
+            String name = components[0];
+
+            String[] arguments = new String[components.length - 1];
+            System.arraycopy(components, 1, arguments, 0, arguments.length);
+
+            Command command = new Command(name, arguments);
+            CommandDispatcher.dispatch(player, command);
+            return;
+        }
+
+        if (mutedNames.contains(player.getUsername().toLowerCase())) {
+            player.getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, "You have been muted and, therefor, cannot talk."));
+            return;
+        }
+
+        ChatMessage m = new ChatMessage(player.getPlayerId(), player.getPrivilege(), message);
+        room.getPlayers().stream()
+                .map(PlayerWrapper::getConnection)
+                .forEach(c -> c.send(m));
+    }
+
+    public void mutePlayer(PlayerWrapper issuer, String username) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)